lang/java/reef-tests/src/test/java/org/apache/reef/tests/yarnnodelabel/NodeLabelTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.tests.yarnnodelabel;
+
+import org.apache.reef.client.DriverConfiguration;
+import org.apache.reef.client.DriverLauncher;
+import org.apache.reef.client.LauncherStatus;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.exceptions.BindException;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.tests.TestEnvironment;
+import org.apache.reef.tests.TestEnvironmentFactory;
+import org.apache.reef.util.EnvironmentUtils;
+import org.junit.*;
+
+/**
+ * Tests whether Evaluator allocations requested with a given node label are fulfilled.
+ * Before you run this test, you should configure your YARN cluster to use two partition, "default" and "mylabel".
+ * You can refer to https://issues.apache.org/jira/browse/REEF-1750.
+ *
+ * Application manager starts on the default partition. Then the NodeLabelTestDriver submits a Evaluator that labeled
```

Comment:
submits an EvaluatorRequest labeled with "mylabel" and one labeled with "default"?